src/vivarium/framework/population/manager.py

```diff
@@ -284,22 +287,54 @@ def register_simulant_initializer(
         requires_streams
             The randomness streams necessary to initialize the simulant
             attributes.
+        required_resources
+            The resources that the initializer requires to run. Strings are
+            interpreted as column names, and Pipelines and RandomnessStreams
+            are interpreted as value pipelines and randomness streams,
         """
-        if isinstance(creates_columns, str):
-            creates_columns = [creates_columns]
+
+        has_individual_requires = requires_columns or requires_values or requires_streams
+
+        if has_individual_requires and required_resources:
```

Comment:
Just so I understand, this is _only_ a check b/c we are trying to be backward compatible w/ individual resources, right?

If so, then maybe mention that in the ValueError below that providing individual resources is deprecated

Reply:
I'm not doing that here, because this method should only get called via `component._register_simulant_initializer()`, which already produces a `DeprecationWarning`.